projects/plugins/jetpack/class-jetpack-pre-connection-jitms.php

```diff
@@ -48,11 +39,32 @@ private function get_raw_messages() {
 				'message_path'   => '/wp:edit-post:admin_notices/',
 				'message'        => __( 'Do you know which of these posts gets the most traffic?', 'jetpack' ),
 				'description'    => __( 'Set up Jetpack to get in-depth stats about your content and visitors.', 'jetpack' ),
-				'button_link'    => $jetpack_setup_url,
 				'button_caption' => __( 'Set up Jetpack', 'jetpack' ),
 			);
 		}
 
+		foreach ( $messages as $key => $message ) {
+			/*
+			 * Add Connect URL to each message, with from including jitm id.
+			 */
+			$jetpack_setup_url               = $this->generate_admin_url(
+				array(
+					'page'    => 'jetpack',
```

Comment:
I think it should be `'page' => 'jetpack#/setup'`

If we use it as it's used [here](https://github.com/Automattic/jetpack/blob/add%2Fsegmentation_params_to_registration/projects/plugins/jetpack/class.jetpack-connection-banner.php#L204) (plus the `from` param) it should work as expected.

The Set Up button on the big banner in the initial wp-admin dashboard is intercepted by a JS. But, in the end, it just [redirects the window to the URL I showed above](https://github.com/Automattic/jetpack/blob/add%2Fsegmentation_params_to_registration/projects/plugins/jetpack/_inc/connect-button.js#L59) (adding the `from` param). So I think it's fine is we just point it directly there:

Reply:
👍 this should take care of that:
53fb4b984bd800353da295e351ddcc0ab96912ff